### ISSUE TITLE
Incompatibility between java-nats-streaming & Spark: protobuf versions conflict.

### ISSUE THREAD

**Issue by laugimethods** [nats-io/java-nats-streaming#20]
When a connection to NATS Streaming (0.1.2) is called from a Spark Cluster (1.6.2), the following exception does occur:

```
16/08/30 15:24:04 ERROR util.SparkUncaughtExceptionHandler: Uncaught exception in thread Thread[Thread-13,5,main]
java.lang.NoSuchMethodError: com.google.protobuf.Descriptors$Descriptor.getOneofs()Ljava/util/List;
    at com.google.protobuf.GeneratedMessageV3$FieldAccessorTable.<init>(GeneratedMessageV3.java:1721)
    at io.nats.stan.protobuf.Protocol.<clinit>(Protocol.java:125)
    at io.nats.stan.protobuf.ConnectRequest.internalGetFieldAccessorTable(ConnectRequest.java:81)
    at com.google.protobuf.GeneratedMessageV3.getAllFieldsMutable(GeneratedMessageV3.java:123)
    at com.google.protobuf.GeneratedMessageV3.getAllFields(GeneratedMessageV3.java:199)
    at com.google.protobuf.TextFormat$Printer.print(TextFormat.java:272)
    at com.google.protobuf.TextFormat$Printer.access$400(TextFormat.java:248)
    at com.google.protobuf.TextFormat.print(TextFormat.java:71)
    at com.google.protobuf.TextFormat.printToString(TextFormat.java:118)
    at com.google.protobuf.AbstractMessage.toString(AbstractMessage.java:106)
    at io.nats.stan.ConnectionImpl.connect(ConnectionImpl.java:139)
    at io.nats.stan.ConnectionFactory.createConnection(ConnectionFactory.java:40)
    at com.logimethods.connector.nats.to_spark.NatsStreamingToSparkConnectorImpl.receive(NatsStreamingToSparkConnectorImpl.java:223)
    at com.logimethods.connector.nats.to_spark.NatsToSparkConnector$1.run(NatsToSparkConnector.java:106)
```

The root cause of that issue is probably due to a conflict between the versions of Google Protobuf used by java-nats-streaming (3.0.0) & Spark (2.5.0):
http://stackoverflow.com/questions/34487996/how-can-i-use-proto3-with-hadoop-spark

I didn't find an easy way to solve that issue when the Spark Cluster is made of (public) Docker Images.

Alternatives:
- change the dependency of java-nats-streaming to protobuf version 2.5.0
- (best) use a **_shaded_** version of protobuf, so to avoid any potential incompatibilities with other librairies & frameworks (Spark, Mesos, Yarn, Akka, etc.)

FYI https://github.com/apache/spark/blob/master/pom.xml :

> ```
>   </dependency>
>   
>   <dependency>
>     <groupId>com.google.protobuf</groupId>
>     <artifactId>protobuf-java</artifactId>
>     <version>${protobuf.version}</version>
>     <scope>${hadoop.deps.scope}</scope>
>   </dependency>
> ```

**Comment by laugimethods** [comment]
Maybe (?), make use of https://github.com/square/wire instead of Google Protocol Buffers?

In https://github.com/bigdatagenomics/mango/issues/276 :
> I used two different methods for generating Java code from Protobuf schemas here
https://github.com/heuermh/phenopacket-protobuf
The code generated by Google's stuff is pretty bad (at least at version 3.0.0-beta-3 as used there) compared to what the https://github.com/square/wire library generates; I'm not sure about version compatibility with ga4gh though.

**Comment by ColinSullivan1** [comment]
Fixed with #21.  Thank you for finding and fixing this issue.
